examples/IrisClassification/README.md

```diff
@@ -68,13 +60,51 @@ if the torchserve is already running with a different "model_store" location, en
 
 ## Running prediction based on deployed model
 
-IrisClassification model takes 4 different parameters - sepal length, sepal width, petal length and petal width.These parameters can be passed as tensor. For ex: `[4.4000, 3.0000, 1.3000, 0.2000]`.
+`python create_deployment.py --deployment_name iris_classification_1`
+
+Note:
+MLFlow stores the model signature inside MLmodel file and it is important to pass the MLmodel path as an EXTRA_FILE argument to create_deployment script.
+
+## Validating the model signature and running prediction based on deployed model
+
+IrisClassification model takes 4 different parameters - sepal length, sepal width, petal length and petal width.T
 
 For testing [iris dataset](http://archive.ics.uci.edu/ml/datasets/Iris/), we are going to use a sample input tensor placed in `sample.json` file.
 
-Run the following command to invoke prediction of our sample input, whose output is stored in output.json file.
+Run the following command to invoke prediction of our sample input, whose output is printed in the console.
+
+`python predict.py --input_file_path sample.json`
+
+`mlflow-torchserve-plugin` validates the input data against the model signature saved during the training process. 
+
+Following are the enforcements made during the model signature validation process
+
+1. Number of columns present in the input - In this example, the number of columns is 4.
+2. The column names should match the names specified in the model signature (i.e during training process). The column names for iris classification examples are expected to be `sepal length (cm)`, `sepal width (cm)`, `petal length (cm)`, `petal width (cm)`
+3. The input values must match the datatype specified in model signature - In this example the input values must be `double`
+
+If any of the above mentioned enforcements fails, the validation exception is raised.
+
+`sample.json` is created with align to all the enforcements. 
+
+To know more about the model signature implementation in details, check the `iris_handler.py` and 'mlflow_torchserve/SignatureValidator.py'
+
+The model will classify the flower species based on the input test data as one among the three types. A sample output is shown as below.
+
+![Sample Prediction output](screenshots/git_prediction_success.png)
+
+
+To understand the model signature and its output, the following sample files are created. Run the following command to see the validation errors.
+
+`python predict.py --input_file_path sig_invalid_column_name.json`
+
+When the input column name doesn't match with the model signature, validation exception is thrown as below.
+
+![Sample Validation Column name Error](screenshots/git_ms_invalid_column_name.png)
```

Comment:
Can we just copy & paste the exception message here?
````suggestion
```
<Exception message>
```
````

Reply:
Added the exception message